### PR TITLE
feat: v2 maps api (get calls only)

### PR DESCRIPTION
**Pull request by cmyui**

### CODE REVIEW

**Review by alowave223 — CHANGES_REQUESTED** [review]
We should add most played maps on a server and by a single player as it is used by guweb atm in order to move guweb from v1 to v2

**Comment by cmyui** [comment]
> We should add most played maps on a server and by a single player as it is used by guweb atm in order to move guweb from v1 to v2

this will be a separate api

File: app/api/v2/models/maps.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+
+# input models
+
+
+# output models
+
+
+class Map(BaseModel):
+    server: Literal["osu!", "private"]
```
Comment (cmyui) [inline app/api/v2/models/maps.py:16]: this should be moved down to match db table

File: app/api/v2/maps.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+from app.api.v2.common.responses import Success
+from app.api.v2.models.maps import Map
+from app.repositories import maps as maps_repo
+
+router = APIRouter()
+
+
+@router.get("/maps")
+async def get_maps(
+    server: str,
```
Comment (cmyui) [inline app/api/v2/maps.py:20]: this should be moved down to match db table; also should test if typing.Literal can/should be used here

Comment (cmyui) [reply app/api/v2/maps.py]: avoiding for now - there are other similar things happening elsewhere, might standardize these to use enums later.

**Pull request merged**